server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -2607,6 +2608,14 @@ private Pair<List<StoragePoolJoinVO>, Integer> searchForStoragePoolsInternal(Lis
                 throw new InvalidParameterValueException("Invalid scope type: " + cmd.getScope());
             }
         }
+        StoragePoolStatus state = null;
+        if (cmd.getState() != null && EnumUtils.isValidEnumIgnoreCase(StoragePoolStatus.class, cmd.getState())) {
```

Comment:
I think with this flow, state will be skipped if invalid value is passed, it won't enter catch since we are validating before getting the value.
How about validating the value and throw exception if it is invalid (before getEnumIgnoreCase) !

Reply:
@harikrishna-patnala `EnumUtils.getEnumIgnoreCase` returns `null` if the value is invalid. I have added a check for that instead. Let me know if this looks good.

Reply:
thanks @vishesh92